src/components/Card.astro

```diff
@@ -0,0 +1,51 @@
+---
+const { colour, title, description } = Astro.props;
+---
+
+<style define:vars={{ colour }}>
```

Comment:
This is actually really interesting, I didn't consider how and if astro could handle passing code level variables to the style.

Reply:
Yeah, I was thinking how to set this up with as little hustle as possible and found this to be the fastest way to do so. Here ia the [link](https://docs.astro.build/en/reference/directives-reference/#definevars) to Astro docs if you are interested.